### PR TITLE
[QNN] Add support for per channel weight scale in dense op

### PR DESCRIPTION
**Pull request by masahi**
QNN dense op does not accept a vector weight scale as argument at the moment, but this restriction can be fixed trivially. 

please review @anijain2305 @vinx13 @FrozenGene

### CODE REVIEW

File: python/tvm/relay/frontend/tflite.py [hunk 1]
```diff
@@ -982,13 +982,15 @@ def convert_fully_connected(self, op):
...
         weight_expr = self.exp_tab.new_const(weight_value, dtype=weight_tensor_type_str)
+        weight_shape = _infer_shape(weight_expr)
 
         if input_tensor.qnn_params:
             out = _qnn.op.dense(in_expr, weight_expr,
                                 input_zero_point=input_tensor.qnn_params['zero_point'],
                                 kernel_zero_point=weight_tensor.qnn_params['zero_point'],
                                 input_scale=input_tensor.qnn_params['scale'],
                                 kernel_scale=weight_tensor.qnn_params['scale'],
+                                units=weight_shape[1],
```
Comment (masahi) [inline python/tvm/relay/frontend/tflite.py]: can you confirm if this change is correct? This code path is not run during tflite forntend tests. @anijain2305

Comment (anijain2305) [reply python/tvm/relay/frontend/tflite.py]: Should it weight_shape[0]. Dense input shapes are m * k and n * k.
And units means n here, right?

Comment (masahi) [reply python/tvm/relay/frontend/tflite.py]: I think units is supposed to mean output dimension (the dimension that weight scale is applied)

From the test cases in https://github.com/apache/incubator-tvm/blob/70c63829474649f34928d42948bf3e5c7dbfbd75/tests/python/frontend/tflite/test_forward.py#L1381-L1384, I guessed that `weight_shape[1]` is the output dimension.

Comment (masahi) [reply python/tvm/relay/frontend/tflite.py]: fixed

Comment (FrozenGene) [reply python/tvm/relay/frontend/tflite.py]: weight_shape[0] should be the output dimension. The script you see is tf's layout, which is not the same as tflite.

File: tests/python/relay/test_op_qnn_dense.py [hunk 2]
```diff
@@ -75,52 +75,8 @@ def make_configuration(quantized_data,
     return config
 
 
-def make_uint_configuration(use_bias=False, requantize_output=False):
```
Comment (masahi) [inline tests/python/relay/test_op_qnn_dense.py]: This is removed because it is not used in the tests and I get segfault from LLVM when I tried to use it.

Comment (anijain2305) [reply tests/python/relay/test_op_qnn_dense.py]: Yes, we looked into that concluded it is an LLVM bug

**Comment by masahi** [comment]
ready to go @vinx13 @FrozenGene

**Comment by masahi** [comment]
thanks @anijain2305 @FrozenGene @vinx13

**Pull request merged**